### PR TITLE
Update Logger API in C++

### PR DESCRIPTION
**Pull request by bernardnormier**
More string updates, mostly for the Logger API.

### CODE REVIEW

File: cpp/src/Ice/LoggerI.cpp [hunk 1]
```diff
@@ -25,15 +25,15 @@ namespace
     const chrono::minutes retryTimeout = chrono::minutes(5);
 }
 
-Ice::LoggerI::LoggerI(const string& prefix, const string& file, bool convert, size_t sizeMax)
-    : _prefix(prefix),
+Ice::LoggerI::LoggerI(string prefix, const string& file, bool convert, size_t sizeMax)
```
Comment (pepone) [inline cpp/src/Ice/LoggerI.cpp:28]: Shouldn't file be a string too?

File: cpp/src/Ice/OSLogLoggerI.cpp [hunk 2]
```diff
@@ -10,48 +10,48 @@
 using namespace std;
 using namespace Ice;
 
-Ice::OSLogLoggerI::OSLogLoggerI(const std::string& prefix) : _prefix(prefix)
+Ice::OSLogLoggerI::OSLogLoggerI(string prefix) : _prefix(std::move(prefix))
 {
-    const string subsystem = prefix.empty() ? "com.zeroc.ice" : "com.zeroc.ice." + prefix;
+    const string subsystem = _prefix.empty() ? "com.zeroc.ice" : "com.zeroc.ice." + _prefix;
```
Comment (pepone) [inline cpp/src/Ice/OSLogLoggerI.cpp:15]: Not directly related, but we should probably make subsystem a member, we use it in trace too.

File: cpp/src/Ice/SysLoggerI.cpp [hunk 3]
```diff
@@ -12,7 +12,7 @@ using namespace std;
 using namespace Ice;
 using namespace IceInternal;
 
-Ice::SysLoggerI::SysLoggerI(const string& prefix, const string& facilityString) : _facility(0), _prefix(prefix)
+Ice::SysLoggerI::SysLoggerI(string prefix, const string& facilityString) : _facility(0), _prefix(std::move(prefix))
```
Comment (pepone) [inline cpp/src/Ice/SysLoggerI.cpp:15]: Why do we keep facilityString as a reference?

**Review by pepone — APPROVED** [review]

File: cpp/src/Ice/LoggerI.cpp (hunk 1 above)
Comment (bernardnormier) [reply cpp/src/Ice/LoggerI.cpp:28]: Fixed. The logic here was to update the public API but leave internal uses of const string& alone.

File: cpp/src/Ice/SysLoggerI.cpp (hunk 3 above)
Comment (bernardnormier) [reply cpp/src/Ice/SysLoggerI.cpp:15]: Fixed. See earlier comment.

File: cpp/src/Ice/OSLogLoggerI.cpp (hunk 2 above)
Comment (bernardnormier) [reply cpp/src/Ice/OSLogLoggerI.cpp]: Fixed

**Review by externl — APPROVED** [review]
Looks good!

**Pull request merged**